### ISSUE TITLE
Test Failure: `Automattic\VIP\Search\Queue_Test::test_ratelimit_indexing_queue_should_be_populated_if_ratelimiting_enabled`

### ISSUE THREAD

**Issue by sjinks** [Automattic/vip-go-mu-plugins#5546]
```
1) Automattic\VIP\Search\Queue_Test::test_ratelimit_indexing_queue_should_be_populated_if_ratelimiting_enabled
should be 0 occurrences of post id #10 in queue table
Failed asserting that actual size 0 matches expected size 1.

/home/runner/work/vip-go-mu-plugins/vip-go-mu-plugins/tests/search/includes/classes/test-class-queue.php:835
```

Ref: https://github.com/Automattic/vip-go-mu-plugins/actions/runs/8974971947/job/24648525854?pr=5542

**Comment by sjinks** [comment]
Repro:

```sh
./bin/test.sh --order-by=random --random-order-seed=1715025566 --filter=Search --stop-on-failure

**Comment by sjinks** [comment]
Faster repro:

```sh
./bin/test.sh --order-by=random --random-order-seed=1715025566 --filter=Queue_Test --stop-on-failure
```

**Comment by sjinks** [comment]
This passes:

```sh
./bin/test.sh --stop-on-failure --debug --filter=Queue_Test
```

When `test_offload_indexing_to_queue` runs before `test_ratelimit_indexing_queue_should_be_populated_if_ratelimiting_enabled`, everything works:

```sh
./bin/test.sh --stop-on-failure --debug --filter='test_offload_indexing_to_queue|test_ratelimit_indexing_queue_should_be_populated_if_ratelimiting_enabled'
```

**Comment by sjinks** [comment]
```diff
@@ -799,8 +798,13 @@ public function test_ratelimit_indexing_queue_should_be_populated_if_ratelimitin
 		$table_name = $this->queue->schema->get_table_name();
 
 		$sync_manager             = new stdClass();
-		$sync_manager->sync_queue = range( 3, 9 );
+		$sync_manager->sync_queue = [ 1 ];
+
+		$this->queue->offload_indexing_to_queue();
+		$current_bail = apply_filters( 'pre_ep_index_sync_queue', false, $sync_manager, 'post' );
+		$this->assertTrue( $current_bail );
 
+		$sync_manager->sync_queue            = range( 3, 9 );
 		$this->queue::$max_indexing_op_count = 0; // Ensure ratelimiting is enabled
 
 		$this->queue->ratelimit_indexing( true, $sync_manager, 'post' );
```

**Comment by sjinks** [comment]
Fixed in #5547
